clients/pkg/promtail/targets/file/filetarget.go

```diff
@@ -321,32 +334,44 @@ func (t *FileTarget) sync() error {
 	return nil
 }
 
-func (t *FileTarget) startWatching(dirs map[string]struct{}) {
+func (t *FileTarget) startWatching(dirs map[string]struct{}) error {
 	for dir := range dirs {
 		if _, ok := t.getWatch(dir); ok {
 			continue
 		}
 
 		level.Info(t.logger).Log("msg", "watching new directory", "directory", dir)
-		t.targetEventHandler <- fileTargetEvent{
+		select {
+		case <-t.quit:
+			return errFileTargetStopped
+		case t.targetEventHandler <- fileTargetEvent{
 			path:      dir,
 			eventType: fileTargetEventWatchStart,
+		}:
+			// continue
 		}
 	}
+	return nil
 }
 
-func (t *FileTarget) stopWatching(dirs map[string]struct{}) {
+func (t *FileTarget) stopWatching(dirs map[string]struct{}) error {
 	for dir := range dirs {
 		if _, ok := t.getWatch(dir); !ok {
 			continue
 		}
 
 		level.Info(t.logger).Log("msg", "removing directory from watcher", "directory", dir)
-		t.targetEventHandler <- fileTargetEvent{
+		select {
+		case <-t.quit:
+			return errFileTargetStopped
+		case t.targetEventHandler <- fileTargetEvent{
 			path:      dir,
 			eventType: fileTargetEventWatchStop,
+		}:
+			// continue
```

Comment:
the bracket here seems wrong? and we don't need the continue

Reply:
Why would it be wrong? We are sending a `fileTargetEvent` to the channel.

The point of the "continue" is just to indicate that we are continuing the loop with the next item. If you think it's obvious, then I could remove the comment :)

Reply:
:facepalm: my mistake, diff was rendering strangely on my screen because my browser was too small. Yeah, lets remove the comment since we're not explicitly `continue`'ing, nor do we want to.